Makefile

```diff
@@ -12,29 +12,30 @@ test: ## run all unit tests
 gofmt: ## format all go files
 	gofmt -l -s -w .
 
-govet:
+govet: ## run go's code vetting on all code
 	go vet ./...
 
-errcheck:
-	go get github.com/kisielk/errcheck@v1.2.0
-	errcheck -exclude .errcheck-ignore ./...
+ci-lint: ## see https://golangci-lint.run/, applies .golangci.yml
+	golangci-lint run
 
-lint: govet errcheck ## run linter checks
+lint: govet ci-lint
 
-goimports:  ## sort all imports
+goimports: ## sort and separate imports
 	go get golang.org/x/tools/cmd/goimports
 	goimports -l -w .
 
+goclean: gofmt goimports ## apply go style rules to source
+
 regenerate: ## regenerate all example and test files
 	make install
 	make -C example regenerate
 	make -C internal/test regenerate
-	make gofmt
-	make goimports
+	make goclean
 	make lint
 
 check: ## regenerate, lint and run a terse version of check
 	@# quietly install and regenerate
+	@go mod tidy
```

Comment:
I use this as part of my pre-commit workflow, but it should not be part of regenerate because make ci uses regenerate, and it *will* generate ci diffs if we run go mod tidy in the ci build.